src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -192,23 +236,81 @@ public void setNumberOfInstances(int numberOfInstances) {
 
 			List<String> urls = sas.getUrls();
 			if (urls != null && !urls.isEmpty()) {
-				v.setAccessURL(this.determineAccessURL(v.getTarget().getProtocol(), urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns(), v.getTarget().getPath()));
+				// Set the access url to the selected route (without any protocol or path yet)
+				v.setAccessURL(this.determineApplicationRoute(urls, v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));				
+      }      
+           
+			v.setNumberOfInstances(sas.getInstances());
+			
+			return Mono.just(v);
+    });
+        
+    Flux<List<RouteResource>> routeApplicationFlux = osaVectorSpaceFlux.flatMapSequential(v -> this.getAppRoutes(v.getApplicationId()));
```

Comment:
Performance impact: for each application we will trigger a new request here. If we have thousands of applications (which we already saw in an issue!) this will cause a lot of traffic at the CFCC (even if we cache it properly). 
Moreover, as of now, this isn't guarded yet: All consumers (if they like it or not) would hit this thing here. => not a good option...

Reply:
There is an API the returns _all_ routes for a space. This would result in a larger response but many less requests. We could then cache that response.

Its available in V2 api too so that would just leave the domain part in V3.

Sound like a good refactor?